lib/core/test/unit/Cardano/Pool/DB/Properties.hs

```diff
@@ -937,6 +945,28 @@ prop_removePools
         fmap (Set.fromList . fmap (view #poolId . snd) . catMaybes)
             <$> atomically $ mapM readPoolRetirement $ Set.toList pools
 
+-- Tests that the 'removeRetiredPools' operation does not do anything other
+-- than removing the pools that the DB layer indicates as having retired.
+--
+prop_removeRetiredPools
```

Comment:
Isn’t the prop_removeRetiredPools essentially just re-iterating the definition of removeRetiredPools though?

Wouldn't we be as fine without it?

Reply:
> Isn’t the prop_removeRetiredPools essentially just re-iterating the definition of removeRetiredPools though? Wouldn't we be as fine without it?

It's admittedly a very simple mock, in this case.

It also checks that `removeRetiredPools` doesn't do anything else, as calling other operations from `DBLayer` will raise an error.